dags/inference/configs/trt_llm_mlperf_v40_config.py

```diff
@@ -0,0 +1,153 @@
+      "sudo mdadm --detail --prefer=by-id /dev/md0",
+      "sudo mkfs.ext4 -F /dev/md0",
+      "sudo mkdir -p /scratch",
+      "sudo mount /dev/md0 /scratch",
+      "sudo chmod a+w /scratch",
+      "cd /scratch",
+      # Prepare data
+      "mkdir -p models",
+      "mkdir -p preprocessed_data",
+      "gsutil -m cp -n -r gs://tohaowu/mlpinf-v40/models/Llama2 models/",
+      "gsutil -m cp -n -r gs://tohaowu/mlpinf-v40/preprocessed_data/open_orca preprocessed_data/",
+      "mv models/Llama2/fp8-quantized-ammo/llama2-70b-chat-hf-tp2pp1-fp8/ models/Llama2/fp8-quantized-ammo/llama2-70b-tp2pp1-fp8/",
+      "git clone https://github.com/mlcommons/inference_results_v4.0",
+      "cd /scratch/inference_results_v4.0/closed/Google",
+      "export MLPERF_SCRATCH_PATH=/scratch",
+      "cp /scratch/inference_results_v4.0/closed/{NVIDIA,Google}/Makefile.docker",
+      "sed -i '27i\ARCH=x86_64' Makefile",
+      "sed -i '29i\ARCH=x86_64' Makefile.docker",
+      "sudo usermod -a -G docker $USER",
+      # Build and launch a docker container
+      "make prebuild DOCKER_DETACH=1",
+      "make docker_add_user",
+      f"make launch_docker DOCKER_NAME={docker_container_name} DOCKER_ARGS='-d'",
+  )
+
+  jsonl_output_path = "metric_report.jsonl"
+  jsonl_converter_py_lines = (
+      "import sys, json, glob, jsonlines",
+      "metadata_log_pattern = '/scratch/inference_results_v4.0/closed/Google/build/logs/*/*/*/*/metadata.json'",
+      "metadata_log_path = glob.glob(metadata_log_pattern)[0]",
+      "def convert_to_jsonl(json_path, jsonl_path):",
+      "  data = dict()",
+      "  data['dimensions'] = {'framework':'TensorRT-LLM'}",
+      "  data['metrics'] = dict()",
+      "  with open(json_path, 'r') as file:",
+      "      metadatadata = json.load(file)",
+      "      for key in metadatadata:",
+      "          try:",
+      "              float(metadatadata[key])",
+      "              data['metrics'][key] = float(metadatadata[key])",
+      "          except:",
+      "              data['dimensions'][key] = metadatadata[key]",
+      "  with jsonlines.open(jsonl_path, 'w') as writter:",
+      "      writter.write(data)",
+      "if __name__ == '__main__':",
+      "  convert_to_jsonl(metadata_log_path, sys.argv[1])",
+  )
+  py_script = "\n".join(jsonl_converter_py_lines)
+  make_jsonl_converter_cmd = f'echo "{py_script}" > jsonl_converter.py'
+
+  docker_cmds = (
+      "make link_dirs",
+      "make build BUILD_TRTLLM=1",
+      f'make run RUN_ARGS="--benchmarks={model_configs["model_name"]} --scenarios=Offline"',
```

Comment:
What batch size does this run? Do you plan to run with different permutations of batch size, number of chips, quantization, etc.? (If so, we can add in a always add in different PR). Just curious.

Reply:
Maybe it's already encoded in the `fp8-quantized-ammo/llama2-70b-chat-hf-tp2pp1-fp8` name. What does ammo and tp2pp1 mean?

Reply:
For A3(and A3+) the config is settled to MLPerf4.0 tuned (batch size is [16 by default]( https://github.com/mlcommons/inference_results_v4.0/blob/ceef1eab947e15915d3208c9ce8c52ed418dd1f7/closed/Google/code/llama2-70b/tensorrt/builder.py#L58)) . For future use cases for G2 and A2 the parameters would be tuned so probably need some similar sweeping logic. 
The names are set here https://github.com/mlcommons/inference_results_v4.0/blob/ceef1eab947e15915d3208c9ce8c52ed418dd1f7/closed/Google/code/llama2-70b/tensorrt/builder.py#L53 @tohaowu might know more about what the names means.